code/game/machinery/air_alarm.dm

```diff
@@ -1081,26 +1090,56 @@ table tr:first-child th:first-child { border: none;}
 		to_chat(user, "The circuit is missing.")
 
 
+/obj/machinery/alarm/north
+	dir = NORTH
+
+/obj/machinery/alarm/east
+	dir = EAST
+
+/obj/machinery/alarm/west
+	dir = WEST
 
 
 /obj/machinery/alarm/monitor
-	apply_danger_level = 0
-	breach_detection = 0
-	post_alert = 0
+	apply_danger_level = FALSE
+	breach_detection = FALSE
+	post_alert = FALSE
 
-/obj/machinery/alarm/server/New()
-	..()
+/obj/machinery/alarm/monitor/north
+	dir = NORTH
+
+/obj/machinery/alarm/monitor/east
+	dir = EAST
+
+/obj/machinery/alarm/monitor/west
+	dir = WEST
+
+
+/obj/machinery/alarm/server
 	req_one_access = list(ACCESS_CIVILIAN_ENGINEERING)
+	target_temperature = 90
+
+/obj/machinery/alarm/server/first_run()
+	alarm_area = get_area(src)
```

Comment:
Put this after an `else`, under `if(alarm_area.master)`

Reply:
The area is being set here. without this, it'd read null.master.
Mostly a copypasta of the parent's otherwise.